src/t8_schemes/t8_standalone/t8_standalone_implementation.hxx

```diff
@@ -0,0 +1,1568 @@
+      maxexclor |= (elem1->coords[idim] ^ elem2->coords[idim]);
+    }
+
+    int num_zeros = t8_standalone_scheme<TEclass>::number_of_leading_zeros (maxexclor);
+    /* If one element already is the ancestor of the other element num_zeros evaluates to maxlevel, in that case return the coarser of both levels*/
+    int level = SC_MIN (num_zeros, (int) SC_MIN (elem1->level, elem2->level));
+    return level;
+  }
+
+  static inline int
+  number_of_leading_zeros (t8_element_coord maxexclor)
+  {
+    int num_of_active_bits_used = SC_LOG2_32 (maxexclor) + 1;
+    T8_ASSERT (num_of_active_bits_used <= T8_ELEMENT_MAXLEVEL[TEclass]);
+
+    int num_zeros = T8_ELEMENT_MAXLEVEL[TEclass] - num_of_active_bits_used;
+    return num_zeros;
+  }
+
+  /**
+ * Set the \a shift last bits of every coordinate to zero. 
+ * 
+ * \param[in, out]  elem     Input element
+ * \param[in]       shift Number of bits to set to zero
+ */
+  static inline void
+  element_cut_coordinates (t8_standalone_element<TEclass> *elem, const int shift)
+  {
+    T8_ASSERT (0 <= shift && shift <= T8_ELEMENT_MAXLEVEL[TEclass]);
+    for (int i = 0; i < T8_ELEMENT_DIM[TEclass]; i++) {
+      elem->coords[i] = (elem->coords[i] >> shift) << shift;
+    }
+  }
+
+  /**
+ * Set the least significant coordinates bits to zero. 
+ * 
+ * \param[in]  elem        Input element
+ * \param[in, out]       parent_elem Parent element
+ * \param[in]       length      int that is 1 at the level of the input element
+ * Note length is used as additional input to avoid recomputation. 
+ */
+  static inline void
+  set_coords_at_level_to_zero (const t8_standalone_element<TEclass> *elem, t8_standalone_element<TEclass> *parent_elem,
+                               const t8_element_coord length)
+  {
+    for (int i = 0; i < T8_ELEMENT_DIM[TEclass]; i++) {
+      parent_elem->coords[i] = elem->coords[i] & ~length;
+    }
+  }
+
+  /**
+ * Adjust the coordinates based on the cube ID.
+ * 
+ * \param[in]           parent       Input element
+ * \param[in, out]      child     Output element
+ * \param[in]           length   int that is 1 at the level of the child element 
+ * \param[in]           cube_id  Cube ID for bitwise operation
+ * Note length is used as additional input to avoid recomputation. 
+ */
+  static inline void
+  put_cube_id_at_level (const t8_standalone_element<TEclass> *parent, t8_standalone_element<TEclass> *child,
+                        const t8_element_coord length, t8_cube_id cube_id)
+  {
+    for (int i = 0; i < T8_ELEMENT_DIM[TEclass]; i++) {
+      child->coords[i] = parent->coords[i] + ((cube_id & (1 << i)) ? length : 0);
+    }
+  }
+
+  static inline t8_element_coord
+  get_root_len ()
+  {
+    if constexpr (TEclass == T8_ECLASS_VERTEX) {
+      return 0;
+    }
+    else {
+      return 1 << T8_ELEMENT_MAXLEVEL[TEclass];
+    }
+  }
+
+  static inline t8_linearidx_t
+  num_descendants_at_leveldiff (const t8_element_t *elem, const int leveldiff)
```

Comment:
if we make leveldiff an unsigend int we could remove the first if and make it a constexpr.